fastf1/api.py

```diff
@@ -318,16 +322,47 @@ def _write_cache(cls, data, cache_file_path, **kwargs):
             pickle.dump(new_cached, cache_file_obj)
 
     @classmethod
-    def _show_not_enabled_warning(cls):
-        if not cls._CACHE_DIR and not cls._has_been_warned:
-            # warn only once and only if cache is not enabled
-            logging.warning(
-                "\n\nNO CACHE! Api caching has not been enabled! \n\t"
-                "It is highly recommended to enable this feature for much "
-                "faster data loading!\n\t"
-                "Use `fastf1.Cache.enable_cache('path/to/cache/')`\n")
+    def _enable_default_cache(cls):
+        if not cls._CACHE_DIR and not cls._default_cache_enabled:
+            cache_dir = None
+            if "FASTF1_CACHE" in os.environ:
+                cache_dir = os.environ.get("FASTF1_CACHE")
+            else:
+                if sys.platform == "linux":
+                    # If .cache exists we will use it. Otherwise, ~/
+                    tmp = os.path.expanduser("~/.cache")
+                    if os.path.exists(tmp):
+                        cache_dir = r"~/.cache/fastf1"
+                    else:
+                        cache_dir = r"~/.fastf1"
+                elif sys.platform == "darwin":
+                    cache_dir = r"~/Library/Caches/fastf1"
+                elif sys.platform == "win32":
+                    cache_dir = r"%LOCALAPPDATA%\Temp\fastf1"
+
+            if cache_dir is not None:
+                # Ensure the default cache folder exists
+                cache_dir = os.path.expandvars(cache_dir)
+                cache_dir = os.path.expanduser(cache_dir)
+                if not os.path.exists(cache_dir):
+                    os.mkdir(cache_dir, mode=0o0700)
```

Comment:
We should use some kind of improved error handling here, if the directory creation fails. I'd say just catch the error and re-raise it, but state specifically that the cache directory creation failed and state which path it was trying to create.